refactor(linter): rename `plugin_name` to `plugin_specifier`

overlookmotel opened the pull request:
Follow-on after #11980. Pure refactor. Rename vars from `plugin_name` to `plugin_specifier`, because that better describes what they are. Later on, we'll add code which does deal with actual plugin names, at which point it'll become confusing.

Sysix commented on crates/oxc_linter/src/config/config_builder.rs:
```diff
@@ -406,15 +406,15 @@ impl ConfigStoreBuilder {
         use crate::PluginLoadResult;
 
-        let resolved = resolver.resolve(oxlintrc_dir_path, plugin_name).map_err(|e| {
+        let resolved = resolver.resolve(oxlintrc_dir_path, plugin_relative_path).map_err(|e| {
```
Is this really a path? I hoped that package.json entries will be resolved too.

overlookmotel replied:
It can be either e.g. `my-plugin` or `../my-plugin.js`.

But you're right - "plugin name" isn't right, but "plugin relative path" also isn't the ideal description.

What is the right word? "specifier"?

overlookmotel replied:
I've gone with "specifier". Shout if you can think of a better name.

camc314 commented:
### Merge activity

* **Jul 9, 8:53 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(linter): rename `plugin_name` to `plugin_specifier` (#12148)

Follow-on after #11980. Pure refactor. Rename vars from `plugin_name` to `plugin_specifier`, because that better describes what they are. Later on, we'll add code which does deal with actual plugin names, at which point it'll become confusing.
